src/core/Microsoft.PowerToys.Settings.UI/ViewModels/GeneralViewModel.cs

```diff
@@ -32,6 +34,18 @@ public GeneralViewModel()
             try
             {
                 GeneralSettingsConfigs = SettingsUtils.GetSettings<GeneralSettings>(string.Empty);
+
+                if (Helper.CompareVersions(GeneralSettingsConfigs.PowertoysVersion, Helper.GetProductVersion()) < 0)
+                {
+                    // Update settings
+                    GeneralSettingsConfigs.PowertoysVersion = Helper.GetProductVersion();
+                    SettingsUtils.SaveSettings(GeneralSettingsConfigs.ToJsonString(), string.Empty);

```

Comment:
why not do this in the ViewModel?

Reply:
What do you mean specifically? What ViewModel?